### PR TITLE
Fix #80: Prepare consent form interface and sample implementation

### LINKED ISSUE #80: Prepare consent form interface and sample implementation
The issue is related to implementation of OAuth 2.0 consent screen in Web Flow:
https://github.com/wultra/powerauth-webflow/issues/461

We should prepare interface changes and sample (mock) implementation of consent screen in Data Adapter - consent form creation, validation and persistence of selected options.

### PR DESCRIPTION
**Pull request by romanstrobl**
I am opening a draft pull request to review the implementation together with the main pull request:
https://github.com/wultra/powerauth-webflow/pull/514

### CODE REVIEW

**Comment by romanstrobl** [comment]
The reason for the funky code was to use the `MethodArgumentNotValidException` class which is used by Spring validation framework, but if we are ok with a different exception class, then it is not needed.

**Comment by petrdvorak** [comment]
OK then - I am OK with it...

**Comment by romanstrobl** [comment]
I may have misunderstood how `@Valid` works, it shouldn't be necessary to check the result manually as we do now.

**Comment by romanstrobl** [comment]
Ok, I think I see what is the problem. Spring does not throw a 400 related exception when BindingResult is present as a method parameter.

File: powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/controller/ConsentController.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+     * @param request Create consent form request.
+     * @return Create consent form response.
+     * @throws DataAdapterRemoteException In case communication with remote system fails.
+     * @throws InvalidOperationContextException In case operation context is invalid.
+     */
+    @RequestMapping(value = "/create", method = RequestMethod.POST)
+    public ObjectResponse<CreateConsentFormResponse> createConsentForm(@Valid @RequestBody ObjectRequest<CreateConsentFormRequest> request, BindingResult result) throws MethodArgumentNotValidException, DataAdapterRemoteException, InvalidOperationContextException {
+        if (result.hasErrors()) {
+            // Call of getEnclosingMethod() on local class returns a reference to current method
+            class Local {}
```
[the remark this answers is not in the document]
Comment (romanstrobl) [reply powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/controller/ConsentController.java]: Fixed by removing the `BindingResult` parameter and manual validation code.

File: powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/controller/ConsentController.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+     * @param request Validate consent form request.
+     * @return Validate consent form response.
+     * @throws DataAdapterRemoteException In case communication with remote system fails.
+     * @throws InvalidOperationContextException In case operation context is invalid.
+     */
+    @RequestMapping(value = "/validate", method = RequestMethod.POST)
+    public ObjectResponse<ValidateConsentFormResponse> validateConsentForm(@Valid @RequestBody ObjectRequest<ValidateConsentFormRequest> request, BindingResult result) throws MethodArgumentNotValidException, DataAdapterRemoteException, InvalidOperationContextException, InvalidConsentDataException {
+        if (result.hasErrors()) {
+            // Call of getEnclosingMethod() on local class returns a reference to current method
+            class Local {}
```
[the remark this answers is not in the document]
Comment (romanstrobl) [reply powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/controller/ConsentController.java]: dtto

File: powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/controller/ConsentController.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+     * @param request Save consent form request.
+     * @return Save consent form response.
+     * @throws DataAdapterRemoteException In case communication with remote system fails.
+     * @throws InvalidOperationContextException In case operation context is invalid.
+     */
+    @RequestMapping(value = "/save", method = RequestMethod.POST)
+    public ObjectResponse<SaveConsentFormResponse> saveConsentForm(@Valid @RequestBody ObjectRequest<SaveConsentFormRequest> request, BindingResult result) throws MethodArgumentNotValidException, DataAdapterRemoteException, InvalidOperationContextException {
+        if (result.hasErrors()) {
+            // Call of getEnclosingMethod() on local class returns a reference to current method
+            class Local {}
```
[the remark this answers is not in the document]
Comment (romanstrobl) [reply powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/controller/ConsentController.java]: dtto

**Comment by romanstrobl** [comment]
I removed all the validation code, it is handled automatically by Spring now.

**Comment by petrdvorak** [comment]
Looks OK

**Pull request merged**